zebra-network/src/peer_set/initialize.rs

```diff
@@ -211,7 +211,18 @@ where
     S: Service<(TcpStream, SocketAddr), Response = peer::Client, Error = BoxError> + Clone,
     S::Future: Send + 'static,
 {
-    let listener = TcpListener::bind(addr).await?;
+    let check_listener = TcpListener::bind(addr).await;
+
+    let listener = match check_listener {
```

Comment:
I found this variable name confusing - I thought we were opening an extra listener to check that the port was available. But it is actually just the result of the listener bind.

```suggestion
    let listener_result = TcpListener::bind(addr).await;

    let listener = match listener_result {
```